core/ops.rs

```diff
@@ -173,9 +175,10 @@ impl OpCtx {
     *opt_mut = Some(error);
   }
 
-  /// Get the [`ContextState`] for this op.
-  pub(crate) fn context_state(&self) -> &ContextState {
-    &self.context_state
+  // TODO(bartlomieju): this forces use of `DefaultOpDriver`, but making it
+  // generic is a bit cumbersome. Probably need to address before landing.
+  pub(crate) fn op_driver(&self) -> &DefaultOpDriver {
+    &self.op_driver
```

Comment:
This is the crux of the fix - we only needed to have `ContextState` here so we could dispatch ops. But we can just pass "op driver" instead. 

This needs to be fixed before landing, because it (incorrectly) defines that we're using `DefaultOpDriver`.

Requesting help.

Reply:
Actually maybe using `DefaultOpDriver` is fine here? Since we give no way to switch driver besides Cargo features - and switching a feature will just switch out the default driver.

Reply:
Maybe it should be named `RealOpDriver`?

Reply:
Went with `RealOpDriver`.